src/tools/PICS-generator/PICSGenerator.py

```diff
@@ -0,0 +1,429 @@
+
+    # Map clusters to common XML template if needed
+    otaProviderCluster = "OTA Software Update Provider Cluster"
+    otaRequestorCluster = "OTA Software Update Requestor Cluster"
+    onOffCluster = "On/Off Cluster"
+    groupKeyManagementCluster = "Group Key Management Cluster"
+    nodeOperationalCredentialsCluster = "Node Operational Credentials Cluster"
+    basicInformationCluster = "Basic Information Cluster"
+    networkCommissioningCluster = "Network Commissioning Cluster"
+
+    if otaProviderCluster in clusterName or otaRequestorCluster in clusterName:
+        clusterName = "OTA Software Update"
+
+    elif onOffCluster == clusterName:
+        clusterName = clusterName.replace("/", "-")
+
+    elif groupKeyManagementCluster == clusterName:
+        clusterName = "Group Communication"
+
+    elif nodeOperationalCredentialsCluster == clusterName or basicInformationCluster == clusterName or networkCommissioningCluster == clusterName:
+        clusterName = clusterName.replace("Cluster", "").strip()
+
+    # Determine if file has already been handled and use this file
+    for outputFolderFileName in os.listdir(outputPathStr):
+        if clusterName in outputFolderFileName:
+            xmlPath = outputPathStr
+            fileName = outputFolderFileName
+            break
+
+    # If no file is found in output folder, determine if there is a match for the cluster name in input folder
+    if fileName == "":
+        for file in xmlFileList:
+            if file.lower().startswith(clusterName.lower()):
+                fileName = file
+                break
+        else:
+            console.print(f"[red]Could not find matching file for \"{clusterName}\" ❌")
+            return
+
+    try:
+        # Open the XML PICS template file
+        console.print(f"Open \"{xmlPath}{fileName}\"")
+        parser = ET.XMLParser(target=ET.TreeBuilder(insert_comments=True))
+        tree = ET.parse(f"{xmlPath}{fileName}", parser)
+        root = tree.getroot()
+    except ET.ParseError:
+        console.print(f"[red]Could not find \"{fileName}\" ❌")
+        return
+
+    # Usage PICS
+    # console.print(clusterPicsCode)
+    usageNode = root.find('usage')
+    for picsItem in usageNode:
+        itemNumberElement = picsItem.find('itemNumber')
+
+        console.print(f"Searching for {itemNumberElement.text}")
+
+        if itemNumberElement.text == f"{clusterPicsCode}":
+            console.print("Found usage PICS value in XML template ✅")
+            supportElement = picsItem.find('support')
+            # console.print(f"Support: {supportElement.text}")
+            supportElement.text = "true"
+
+            # Since usage PICS (server or client) is not a list, we can break out when a match is found,
+            # no reason to keep iterating through the elements.
+            break
+
+    # Feature PICS
+    # console.print(featurePicsList)
+    featureNode = root.find("./clusterSide[@type='Server']/features")
```

Comment:
Is the root here the cluster root or the xml root? Ie, does this work if there is > 1 cluster in an xml file?

Reply:
ditto for the find commands below

Reply:
The PICS XML files are per cluster, so there should not be multiple clusters in the templates 🤔 At least I have not noticed a XML which has that. But true, in the case we do have that I need to adjust the code to handle that.

Reply:
I think I saw this in the ota one,but the DM XML is broken right now. Either way, easy fix, let's do it in a follow up. There's also the issue of aliased clusters, which is probably also a followup.

Reply:
The OTA one is strange and in it current for does really not cover that much. Actually, looking at it again, this is kind of the same as with GroupKey, it does not really check the clusters, e.g. there is no 1.1 test case for the OTA clusters, but I guess we get these covered by the latest IDM test now.

Reply:
The way the aliased clusters ended up being handled is that they have they own PICS, hence their own PICS XML file, so at least they should not be an issue.

Reply:
In the current PICS XML files, this should not be a problem. In case we find one which causes issues, I will be ready to address this, I have resolved this for now. I hope that is ok 😄